GOA v2 -- Leaving off the Token field for an OAuth method causes a panic

When creating a new method that required a `Token`, when I accidentally forgot to include it in the payload, `goa gen` caused a hard to diagnose panic.

Here is a method that will crash (this is edited from https://github.com/goadesign/examples/blob/34577a438e956c1904238d8c4bdb5f704fb33d7c/security/design/design.go#L96)

```
	Method("secure", func() {
		Description("This action is secured with the jwt scheme")

		Security(JWTAuth, func() { // Use JWT to auth requests to this endpoint.
			Scope("api:read") // Enforce presence of "api:read" scope in JWT claims.
		})

		Payload(func() {
                        // MISSING FIELD FOR TOKEN HERE.
			Required("token")
		})

		Result(String)

		Error("invalid-scopes", String, "Token scopes are invalid")

		HTTP(func() {
			GET("/secure")
			Response(StatusOK)
			Response("invalid-scopes", StatusForbidden)
		})
	})
```

## Expected Behavior

To receive an error code indicating the problem, such as:

`service "secured_service" method "doubly_secure": payload - required field "token" does not exist in type object`

## Actual Behavior
Panic

```
exit status 2
panic:  is not the name of a child of the mapped attribute

goroutine 1 [running]:
goa.design/goa/expr.(*MappedAttributeExpr).Map(0xc000215320, 0xa8bd42, 0xd, 0x0, 0x0)
        /home/james/go/src/goa.design/goa/expr/mapped_attribute.go:124 +0x19d
goa.design/goa/expr.(*HTTPEndpointExpr).Finalize(0xc0002a2580)
        /home/james/go/src/goa.design/goa/expr/http_endpoint.go:478 +0x85a
goa.design/goa/eval.finalizeSet(0xc0001f77b0, 0x1, 0x1, 0x0, 0x0)
        /home/james/go/src/goa.design/goa/eval/eval.go:233 +0xae
goa.design/goa/expr.(*RootExpr).WalkSets(0xfbdde0, 0xaec558)
        /home/james/go/src/goa.design/goa/expr/root.go:121 +0xab2
goa.design/goa/eval.RunDSL(0x7ffe98c2818c, 0xc0001f4320)
        /home/james/go/src/goa.design/goa/eval/eval.go:55 +0x48e
main.main()
        /home/james/go/src/goa.design/examples/basic/goa753429668/main.go:47 +0x411
```

## Steps to Reproduce the Problem (including complete and simple design if relevant)
See comments below for the whole design file, then use Goa V2 to generate code.

## Specifications

  - goa Version: V2

> commit f1b9b0ebaef41459898c247dbc5a0dd7acea3d03 (HEAD -> v2, origin/v2, origin/master, origin/HEAD)
> Author: Nitin <nitinmohan87@gmail.com>
> Date:   Thu Apr 25 12:18:16 2019 -0700
> 
>    Add support for Params and Headers DSL (#2077)

  - Platform (Linux/Windows etc.): Ubuntu Linux 18.04.2 LTS running go1.12.2

--- a/expr/http_endpoint.go
+++ b/expr/http_endpoint.go
@@ -228,7 +228,6 @@ func (e *HTTPEndpointExpr) Prepare() {
 // Validate validates the endpoint expression.
 func (e *HTTPEndpointExpr) Validate() error {
 	verr := new(eval.ValidationErrors)
-
 	// Name cannot be empty
 	if e.Name() == "" {
 		verr.Add(e, "Endpoint name cannot be empty")
@@ -321,6 +320,29 @@ func (e *HTTPEndpointExpr) Validate() error {
 	// Validate errors
 	for _, er := range e.HTTPErrors {
 		verr.Merge(er.Validate())
+	}
+
+	// Validate security definitions
+	for _, req := range e.MethodExpr.Requirements {
+		for _, sch := range req.Schemes {
+			var name, msg string
+			switch sch.Kind {
+			case APIKeyKind:
+				name = "security:apikey:" + sch.SchemeName
+				msg = "API key attribute (use APIKey)"
+			case JWTKind:
+				name = "security:token"
+				msg = "JWT token attribute (use Token)"
+			case OAuth2Kind:
+				name = "security:accesstoken"
+				msg = "access token attribute (use AccessToken)"
+			}
+			if name != "" {
+				if f := TaggedAttribute(e.MethodExpr.Payload, name); f == "" {
+					verr.Add(e, "Payload must define %s required by security scheme %q.", msg, sch.SchemeName)
+				}
+			}
+		}
 	}
 
 	// Validate definitions of params, headers and bodies against definition of payload

--- a/expr/http_endpoint_test.go
+++ b/expr/http_endpoint_test.go
@@ -51,6 +51,12 @@ func TestHTTPEndpointValidation(t *testing.T) {
 		"endpoint-body-as-user-type": {
 			DSL: testdata.EndpointBodyAsUserType,
 		},
+		"endpoint-missing-token": {
+			DSL: testdata.EndpointMissingToken,
+			Errors: []string{
+				"service \"Service\" method \"Method\": payload of method \"Method\" of service \"Service\" does not define a JWT attribute, use Token to define one",
+			},
+		},
 	}
 	for name, c := range cases {
 		t.Run(name, func(t *testing.T) {

--- a/expr/testdata/endpoint_dsls.go
+++ b/expr/testdata/endpoint_dsls.go
@@ -103,6 +103,27 @@ var EndpointBodyAsUserType = func() {
 	})
 }
 
+var EndpointMissingToken = func() {
+	var Entity = Type("Entity", func() {
+		Attribute("id", String)
+		Attribute("name", String)
+	})
+	var JWT = JWTSecurity("JWT", func() {
+		Scope("api:read", "Read access")
+	})
+	Service("Service", func() {
+		Security(JWT, func() {
+			Scope("api:read")
+		})
+		Method("Method", func() {
+			Payload(Entity)
+			HTTP(func() {
+				POST("/{id}")
+			})
+		})
+	})
+}
+
 var FinalizeEndpointBodyAsExtendedTypeDSL = func() {
 	var EntityData = Type("EntityData", func() {
 		Attribute("name", String)

